src/java.base/share/classes/java/util/zip/Inflater.java

```diff
@@ -636,14 +636,19 @@ public int getAdler() {
     }
 
     /**
-     * Returns the total number of compressed bytes input so far.
+     * Returns the total number of compressed bytes input so far, discarding
+     * the 32 highest order bits, as if the {@link #getBytesRead()}
+     * method was called followed by a narrowing conversion from <code>long</code>
+     * to <code>int</code>.
      *
-     * <p>Since the number of bytes may be greater than
-     * Integer.MAX_VALUE, the {@link #getBytesRead()} method is now
-     * the preferred means of obtaining this information.</p>
+     * @deprecated This method cannot safely return a result without a potential
+     * loss of information about the magnitude and sign of the returned value.
+     * Use the {@link #getBytesRead()} method instead when obtaining this information.
      *
-     * @return the total number of compressed bytes input so far
+     * @return the total number of compressed bytes input so far, discarding
+     * the 32 highest order bits
      */
+    @Deprecated
```

Comment:
Hello Eirik, the `Deprecated` annotation allows for a `since` property. I think we should use it here `@Deprecated(since=23)`.

Reply:
Fixed.